extensions/src/AWSSDK.Extensions.CborProtocol/Internal/CborStreamReader.cs

```diff
@@ -200,14 +193,13 @@ private void ReadEndContainer(CborContainerType expectedType, CborReaderState ex
 
                 if (state == CborReaderState.Finished)
                 {
+                    // Try to refill first, maybe the break marker is in the next chunk.
+                    RefillBuffer(0);
+
                     if (IsNextByteEndOfContainer())
                     {
                         RefillBuffer(1); // Skip the break marker (0xFF)
```

Comment:
Add a comment about calling `RefillBuffer` a second time on a already filled buffer to trigger an expansion of the size of the buffer for the reasons describe in the comment you will add for line 199.